Modules/LDAPObject.c

```diff
@@ -1193,27 +1193,19 @@ l_ldap_result4(LDAPObject *self, PyObject *args)
     pmsg =
         LDAPmessage_to_python(self->ldap, msg, add_ctrls, add_intermediates);
 
-    if (res_type == 0) {
-        result_str = Py_None;
-        Py_INCREF(Py_None);
-    }
-    else {
-        result_str = PyInt_FromLong(res_type);
-    }
-
     if (pmsg == NULL) {
         retval = NULL;
     }
     else {
         /* s handles NULL, but O does not */
         if (add_extop) {
-            retval = Py_BuildValue("(OOiOsO)", result_str, pmsg, res_msgid,
+            retval = Py_BuildValue("(iOiOsO)", res_type, pmsg, res_msgid,
```

Comment:
This changes the return type for all the `result` functions, right?

Reply:
No change in type here, at this point `result_str` is always positive so we never even returned `None`. Then it might as well be converted to `int()` in `Py_BuildValue` rather than manually...